vulture/config.py

```diff
@@ -0,0 +1,168 @@
+        exclude=vulture_settings.get("exclude", []),
+        ignore_decorators=vulture_settings.get("ignore_decorators", []),
+        ignore_names=vulture_settings.get("ignore_names", []),
+        make_whitelist=vulture_settings.get("make_whitelist", False),
+        min_confidence=vulture_settings.get(
+            "min_confidence", MIN_CONFIDENCE_DEFAULT
+        ),
+        sort_by_size=vulture_settings.get("sort_by_size", False),
+        verbose=vulture_settings.get("verbose", False),
+    )
+    return output
+
+
+def _parse_args(args=None):
+    # type: (Optional[List[str]]) -> Config
+    """
+    Parse CLI arguments
+
+    :param args: A list of strings representing the CLI arguments. If left to
+        the default, this will default to sys.argv
+    """
+
+    def csv(exclude):
+        return exclude.split(",")
+
+    usage = "%(prog)s [options] PATH [PATH ...]"
+    version = "vulture {}".format(__version__)
+    glob_help = "Patterns may contain glob wildcards (*, ?, [abc], [!abc])."
+    parser = argparse.ArgumentParser(prog="vulture", usage=usage)
+    parser.add_argument(
+        "paths",
+        nargs="+",
+        metavar="PATH",
+        help="Paths may be Python files or directories. For each directory"
+        " Vulture analyzes all contained *.py files.",
+    )
+    parser.add_argument(
+        "--exclude",
+        metavar="PATTERNS",
+        type=csv,
+        help="Comma-separated list of paths to ignore (e.g.,"
+        ' "*settings.py,docs/*.py"). {glob_help} A PATTERN without glob'
+        " wildcards is treated as *PATTERN*.".format(**locals()),
+    )
+    parser.add_argument(
+        "--ignore-decorators",
+        metavar="PATTERNS",
+        type=csv,
+        help="Comma-separated list of decorators. Functions and classes using"
+        ' these decorators are ignored (e.g., "@app.route,@require_*").'
+        " {glob_help}".format(**locals()),
+    )
+    parser.add_argument(
+        "--ignore-names",
+        metavar="PATTERNS",
+        type=csv,
+        default=None,
+        help='Comma-separated list of names to ignore (e.g., "visit_*,do_*").'
+        " {glob_help}".format(**locals()),
+    )
+    parser.add_argument(
+        "--make-whitelist",
+        action="store_true",
+        help="Report unused code in a format that can be added to a"
+        " whitelist module.",
+    )
+    parser.add_argument(
+        "--min-confidence",
+        type=int,
+        default=MIN_CONFIDENCE_DEFAULT,
+        help="Minimum confidence (between 0 and 100) for code to be"
+        " reported as unused.",
+    )
+    parser.add_argument(
+        "--sort-by-size",
+        action="store_true",
+        help="Sort unused functions and classes by their lines of code.",
+    )
+    parser.add_argument("-v", "--verbose", action="store_true")
+    parser.add_argument("--version", action="version", version=version)
+    namespace = parser.parse_args(args)
+    output = Config(
```

Comment:
No need for variable `output` (here and above).

Reply:
Addressed in fc7d3fa